### PR TITLE
Geo

### PR DESCRIPTION
**Pull request by JeromeMBourgeois**
Update of the service allowing to make the list of the communes to add a filter on the name of the returned territories
Update of the service to retrieve information about the territories included in "departement {code}" or "region (code) to add a filter on the name of the returned territories
Creation of the service allowing to make the list of the "collectivités d'outre-mer"
Creation of the service to search for an "collectivités d'outre-mer" by its geographical code
Creation of the service to retrieve information about the territories included in "collectivités d'outre-mer" {code}
Creation of the District services

### CODE REVIEW

File: src/main/java/fr/insee/rmes/api/geo/ConstGeoApi.java [hunk 1]
```diff
@@ -51,6 +54,7 @@
     public static final String PATTERN_ARRONDISSEMENT_MUNICIPAL = "";
 
     public static final String PATTERN_COMMUNE_DESCRIPTION = "Code de la commune (cinq caractères)";
+    public static final String PATTERN_COM_DESCRIPTION= "Code de la collectivite d'outre-mer (trois caractères)";
```
Comment (alicela) [inline src/main/java/fr/insee/rmes/api/geo/ConstGeoApi.java:57]: Missing é to  "collectivité"

File: src/main/java/fr/insee/rmes/api/geo/territoire/CommuneApi.java [hunk 2]
```diff
@@ -46,9 +46,9 @@
     private static final String LITTERAL_PARAMETER_DATE_DESCRIPTION =
         "Filtre pour renvoyer la commune active à la date donnée. Par défaut, c’est la date courante. (Format : 'AAAA-MM-JJ')";
     private static final String LITTERAL_PARAMETER_TYPE_DESCRIPTION = "Filtre sur le type de territoire renvoyé.";
-
+    private static final String LITTERAL_PARAMETER_NAME_DESCRIPTION = "Filtre sur le nom de la commune" ;
     private static final String LITTERAL_CODE_EXAMPLE = "14475";
-
+    private static final String LITTERAL_PARAMETER_COM_DESCRIPTION="Ecrire true pour inclure les collectivités d’outre-mer";
```
Comment (alicela) [inline src/main/java/fr/insee/rmes/api/geo/territoire/CommuneApi.java:51]: `Écrire \"true\"`

File: src/main/java/fr/insee/rmes/api/geo/territoire/CommuneApi.java [hunk 3]
```diff
@@ -218,7 +218,18 @@ public Response getListe(
...
+                value = Constants.PARAMETER_DATE) String date,
+        @Parameter(
+                description = LITTERAL_PARAMETER_NAME_DESCRIPTION,
+                required = false,
+                schema = @Schema(type = Constants.TYPE_STRING, example="Bonnay")) @QueryParam(
+                    value = Constants.PARAMETER_FILTRE) String filtreNom,
+        @Parameter(description = LITTERAL_PARAMETER_COM_DESCRIPTION,
+                required = false,
+                schema = @Schema(type = Constants.TYPE_STRING,example="none")) @QueryParam(
+                    value = Constants.PARAMETER_STRING) String com
```
Comment (alicela) [inline src/main/java/fr/insee/rmes/api/geo/territoire/CommuneApi.java:230]: Change to a boolean

File: src/main/java/fr/insee/rmes/api/geo/territoire/DepartementApi.java [hunk 4]
```diff
@@ -42,7 +42,7 @@
         "Informations sur un departement identifié par son code (deux ou trois caractères)";
     private static final String LITTERAL_RESPONSE_DESCRIPTION = "Departement";
     private static final String LITTERAL_PARAMETER_TYPE_DESCRIPTION = "Filtre sur le type de territoire renvoyé.";
-
+    private static final String LITTERAL_PARAMETER_NAME_DESCRIPTION = "Filtre sur le Filtre sur le nom des territoires renvoyés" ;
```
Comment (alicela) [inline src/main/java/fr/insee/rmes/api/geo/territoire/DepartementApi.java:45]: Too much "Filtre sur le"

File: src/main/java/fr/insee/rmes/modeles/geo/territoire/District.java [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+    public District(
+        String code,
+        String uri,
+        String intitule,
+        String type,
+        String dateCreation,
+        String dateSuppression,
+        IntituleSansArticle intituleSansArticle,
+        String chefLieu) {
```
Comment (alicela) [inline src/main/java/fr/insee/rmes/modeles/geo/territoire/District.java:41]: Remove chefLieu

File: src/main/java/fr/insee/rmes/modeles/geo/territoire/District.java [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+    @JacksonXmlProperty(localName = "Type")
+    @Schema(example = "District")
+    public String getType() {
+        return type;
+    }
+
+    @Override
+    @JacksonXmlProperty(localName = "DateCreation")
+    @Schema(
+        description = "Date de création du district si il n’existait pas au premier COG du 1er janvier 1943",
```
Comment (alicela) [inline src/main/java/fr/insee/rmes/modeles/geo/territoire/District.java:77]: si il => s'il

File: src/main/java/fr/insee/rmes/modeles/geo/territoire/District.java [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+        example = "1943-01-01")
+    public String getDateCreation() {
+        return dateCreation;
+    }
+
+    @Override
+    @JacksonXmlProperty(localName = "DateSuppression")
+    @Schema(description = "Date de suppression du district si il a été supprimée. ", example = "2019-01-01")
```
Comment (alicela) [inline src/main/java/fr/insee/rmes/modeles/geo/territoire/District.java:85]: si il => s'il

File: src/main/resources/request/geographie/getTerritoireByCodeAndDateAndFiltreNomCommune.ftlh [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+					igeo:date ?dateSuppressionChefLieu.
+				}
+				
+				<#if date != "*">
+					FILTER(!BOUND(?dateCreationChefLieu) || ?dateCreationChefLieu <= '${date}'^^xsd:date)
+					FILTER(!BOUND(?dateSuppressionChefLieu) || ?dateSuppressionChefLieu > '${date}'^^xsd:date)
+				</#if>
+				
+			}
+			#blabla#
```
Comment (alicela) [inline src/main/resources/request/geographie/getTerritoireByCodeAndDateAndFiltreNomCommune.ftlh:48]: :-D

File: src/main/resources/request/geographie/getTerritoireByCodeAndDateAndFiltreNomCommune.ftlh [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+</#if>
+			OPTIONAL {
+				?evenementCreation igeo:creation ?uri ;
+					               igeo:date ?dateCreation .
+			}
+			OPTIONAL {
+				?evenementSuppression igeo:suppression ?uri ;
+					                  igeo:date ?dateSuppression.
+			}
+			OPTIONAL {
```
Comment (alicela) [inline src/main/resources/request/geographie/getTerritoireByCodeAndDateAndFiltreNomCommune.ftlh:30]: <#if chefLieu != "none">

File: src/main/java/fr/insee/rmes/api/geo/AbstractGeoApi.java [hunk 10]
```diff
@@ -80,6 +80,14 @@ protected boolean verifyParameterDateIsRightWithHistory(String date) {
     protected String formatValidParameterDateIfIsNull(String date) {
         return (date != null) ? date : DateUtils.getDateTodayStringFormat();
     }
+    
+    protected String formatValidParameterFiltreIfIsNull(String filtreNom) {
+        return (filtreNom != null) ? filtreNom : "*";
+    } 
+    
+    protected String formatValidParameterComIfIsNull(String com) {
```
Comment (alicela) [inline src/main/java/fr/insee/rmes/api/geo/AbstractGeoApi.java:88]: This method is the same as above. Remove it

File: src/main/java/fr/insee/rmes/api/geo/territoire/CollectivitesDOutreMerAPI.java [hunk 11]
```diff
@@ -0,0 +1,188 @@
...
+
+	
+	@Path(ConstGeoApi.PATH_GEO)
+	@Tag(name = ConstGeoApi.TAG_NAME, description = ConstGeoApi.TAG_DESCRIPTION)
+	
+	public class CollectivitesDOutreMerAPI extends AbstractGeoApi {
+
+
+
+	    private static Logger logger = LogManager.getLogger(CollectiviteDOutreMer.class);
```
Comment (alicela) [inline src/main/java/fr/insee/rmes/api/geo/territoire/CollectivitesDOutreMerAPI.java:38]: Change to CollectivitesDOutreMerAPI.class

File: src/main/java/fr/insee/rmes/api/geo/territoire/CollectivitesDOutreMerAPI.java [hunk 12]
```diff
@@ -0,0 +1,188 @@
...
+	                pattern = ConstGeoApi.PATTERN_COM,
+	                type = Constants.TYPE_STRING, example=LITTERAL_CODE_EXAMPLE)) @PathParam(Constants.CODE) String code,
+	        @Parameter(hidden = true) @HeaderParam(HttpHeaders.ACCEPT) String header,
+	        @Parameter(
+	            description = LITTERAL_PARAMETER_DATE_DESCRIPTION,
+	            required = false,
+	            schema = @Schema(type = Constants.TYPE_STRING, format = Constants.FORMAT_DATE)) @QueryParam(
+	                value = Constants.PARAMETER_DATE) String date) {
+
+	        logger.debug("Received GET request for commune {}", code);
```
Comment (alicela) [inline src/main/java/fr/insee/rmes/api/geo/territoire/CollectivitesDOutreMerAPI.java:119]: commune or collectivite d'outre-mer  ?

File: src/main/java/fr/insee/rmes/api/geo/territoire/CollectivitesDOutreMerAPI.java [hunk 13]
```diff
@@ -0,0 +1,188 @@
...
+	                content = @Content(schema = @Schema(type = ARRAY, implementation = Territoire.class)),
+	                description = LITTERAL_RESPONSE_DESCRIPTION)
+	        })
+	    public Response getDescendants(
+	        @Parameter(
+	            description = "code de la commune ou du district (3 caractères)",
+	            required = true,
+	            schema = @Schema(
+	                pattern = ConstGeoApi.PATTERN_COM,
+	                type = Constants.TYPE_STRING, example="987")) @PathParam(Constants.CODE) String code,
```
Comment (alicela) [inline src/main/java/fr/insee/rmes/api/geo/territoire/CollectivitesDOutreMerAPI.java:155]: example=LITTERAL_CODE_EXAMPLE

File: src/main/java/fr/insee/rmes/api/geo/territoire/CollectivitesDOutreMerAPI.java [hunk 14]
```diff
@@ -0,0 +1,188 @@
...
+	        @Parameter(hidden = true) @HeaderParam(HttpHeaders.ACCEPT) String header,
+	        @Parameter(
+	            description ="Filtre pour renvoyer les territoires inclus dans la collectivité d'outre-mer active à la date donnée. Par défaut, c’est la date courante. (Format : 'AAAA-MM-JJ')",
+	            required = false,
+	            schema = @Schema(type = Constants.TYPE_STRING, format = Constants.FORMAT_DATE)) @QueryParam(
+	                value = Constants.PARAMETER_DATE) String date,
+	        @Parameter(
+	            description = LITTERAL_PARAMETER_TYPE_DESCRIPTION+ "( Commune ou District )",
```
Comment (alicela) [inline src/main/java/fr/insee/rmes/api/geo/territoire/CollectivitesDOutreMerAPI.java:163]: " (Commune ou District)"

File: src/main/java/fr/insee/rmes/api/geo/territoire/CollectivitesDOutreMerAPI.java (hunk 12 above)
Comment (JeromeMBourgeois) [reply src/main/java/fr/insee/rmes/api/geo/territoire/CollectivitesDOutreMerAPI.java:119]: collectivite d'outre-mer

File: src/main/java/fr/insee/rmes/api/geo/territoire/CommuneApi.java (hunk 3 above)
Comment (JeromeMBourgeois) [reply src/main/java/fr/insee/rmes/api/geo/territoire/CommuneApi.java:230]: I used an enum instead of a boolean

**Pull request merged**